src/test/java/com/amazon/dlic/auth/http/saml/HTTPSamlAuthenticatorTest.java

```diff
@@ -151,6 +152,63 @@ public void basicTest() throws Exception {
         Assert.assertEquals("horst", jwt.getClaim("sub"));
     }
 
+    @Test
+    public void testMetadataBody() throws Exception {
+        mockSamlIdpServer.setSignResponses(true);
+        mockSamlIdpServer.loadSigningKeys("saml/kirk-keystore.jks", "kirk");
+        mockSamlIdpServer.setAuthenticateUser("horst");
+        mockSamlIdpServer.setEndpointQueryString(null);
+
+        // Note: We need to replace endpoint with mockSamlIdpServer endpoint
+        final String metadataBody = FileHelper.loadFile("saml/metadata.xml")
+                                        .replaceAll("http://localhost:33667/", mockSamlIdpServer.getMetadataUri());
+
+        Settings settings = Settings.builder().put("idp.metadata_body", metadataBody)
+            .put("kibana_url", "http://wherever")
+            .put("idp.entity_id", mockSamlIdpServer.getIdpEntityId())
+            .put("exchange_key", "abc").put("roles_key", "roles").put("path.home", ".").build();
```

Comment:
not sure we can adding roles. Will have to confirm

Reply:
Was following the same Settings as other tests in this file - 
https://github.com/opendistro-for-elasticsearch/security/blob/master/src/test/java/com/amazon/dlic/auth/http/saml/HTTPSamlAuthenticatorTest.java#L125-L127

I can run the tests without .put("roles_key", "roles") since by default if roles_key is not specified it defaults to "roles".